cartographer/mapping/global_trajectory_builder_interface.h

```diff
@@ -58,6 +58,17 @@ class GlobalTrajectoryBuilderInterface {
   virtual void AddSensorData(const sensor::OdometryData& odometry_data) = 0;
   virtual void AddSensorData(
       const sensor::FixedFramePoseData& fixed_frame_pose) = 0;
+
+  // An optional callback which is called when local SLAM finishes processing
```

Comment:
Comments should be kept short and concise. There is the danger that when code changes in the future, comments go out of sync. How about this:
```
  // A callback which is called when local SLAM processed an accumulated
  // 'sensor::RangeData'. If the data was inserted into a submap,
  // reports the assigned 'NodeId', otherwise 'nullptr' if the data was filtered
  // out.
```